ddht/v5_1/network.py

```diff
@@ -540,6 +541,53 @@ async def _monitor_done(
     )
 
 
+@asynccontextmanager
+async def common_network_stream_find_nodes(
+    network: NetworkAPI,
+    node_id: NodeID,
+    endpoint: Endpoint,
+    distances: Collection[int],
+    *,
+    request_id: Optional[bytes] = None,
+) -> AsyncIterator[trio.abc.ReceiveChannel[ENRAPI]]:
+    async def _stream_find_nodes_response(
+        send_channel: trio.abc.SendChannel[ENRAPI],
+    ) -> None:
+        async with network.client.stream_find_nodes(
+            node_id, endpoint, distances=distances, request_id=request_id
+        ) as resp_aiter:
+            async with send_channel:
+                async for response in resp_aiter:
+                    enrs = response.message.enrs
+                    for enr in enrs:
+                        try:
+                            await send_channel.send(enr)
+                        except (trio.BrokenResourceError, trio.ClosedResourceError):
+                            break
+
+    if not distances:
+        raise TypeError("Must provide at least one distance")
+
+    if endpoint is None:
+        endpoint = await network.endpoint_for_node_id(node_id)
```

Comment:
I think you should move this inside of the `_stream_find_nodes_response` it took me a minute to figure out what was going on as this variable *appears* to not be used but you have to scroll back up to the function definition to find it being used inside the context.